src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -221,19 +222,48 @@ class IntelliJPlugin implements Plugin<Project> {
     }
 
     private static void configureInstrumentation(@NotNull Project project, @NotNull IntelliJPluginExtension extension) {
-        LOG.info("Configuring IntelliJ compile tasks")
-        def abstractCompileDependencies = { String taskName ->
-            project.tasks.findByName(taskName).collect {
-                it.taskDependencies.getDependencies(it).findAll { it instanceof AbstractCompile }
-            }.flatten() as Collection<AbstractCompile>
-        }
-        abstractCompileDependencies(JavaPlugin.CLASSES_TASK_NAME).each {
-            it.inputs.property("intellijIdeaDependency", extension.ideaDependency.toString())
-            it.doLast(new IntelliJInstrumentCodeAction(false))
+        if (!extension.instrumentCode || extension.type == 'RS') {
+            return
```

Comment:
I'd added the task anyway and did nothing on `extension.instrumentCode || extension.type == 'RS'`